embassy-stm32/src/i2c/v1.rs

```diff
@@ -4,11 +4,24 @@ use embassy_embedded_hal::SetConfig;
 use embassy_hal_common::into_ref;
 
 use crate::gpio::sealed::AFType;
+use crate::gpio::Pull;
 use crate::i2c::{Error, Instance, SclPin, SdaPin};
 use crate::pac::i2c;
 use crate::time::Hertz;
 use crate::Peripheral;
 
+#[non_exhaustive]
+#[derive(Copy, Clone)]
+pub struct Config {
+    pullup_enable: bool,
```

Comment:
Looking at the nRF HAL, maybe this should be 

```
pub struct Config {
    pub sda_pullup: bool,
    pub scl_pullup: bool,
}
```

this would allow setting pullup individually on the pins + keep the naming the same.

Reply:
Done. I also forgot to make config field `pub` 🙄